jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/sys/DatabaseOps.java

```diff
@@ -198,6 +198,12 @@ public static void compact(DatasetGraphSwitchable container) {
             LOG.debug(String.format("Compact %s -> %s\n", db1.getFileName(), db2.getFileName()));
 
             compact(container, loc1, loc2);
+
+            if ( shouldDeleteOld ) {
+                Path loc1Path = IOX.asPath(loc1);
+                LOG.info("Deleting old database after successful compaction (old db path='" + loc1Path + "'...");
+                IOX.delete(loc1Path);
```

Comment:
Does this suppress errors?

Doing this on Windows has a good chance of failing due to the longstanding JVM bug with deleting memory mapped files

Reply:
`Sys.isWindows`.

On windows, it will work ... but the files do not go away until the JVM exits.

Reply:
how do I handle this?

Reply:
I've updated this to use `Files.walk()...`